src/features/ProfilePage/components/PublicInfo.tsx

```diff
@@ -4,74 +4,101 @@ import { useTranslation } from 'react-i18next';
 
 import { selectMentor, selectUser } from '@/features/Authentication/userSlice';
 import { useAppSelector } from '@/store';
-import { useDebounce } from '@/hooks/useDebounce';
 import { useUpdateMentorMutation } from '@/features/MentorPage/mentorPageApi';
 import { useUpdateUserMutation } from '@/features/Authentication/authenticationApi';
 
-import {
-  DEFAULT_ICON_SIZE,
-  palette,
-  SAVE_DELAY_MS,
-} from '@/components/constants';
+import { ButtonRow } from '.';
+import { DEFAULT_ICON_SIZE, palette } from '@/components/constants';
 import LabeledInput from '@/components/LabeledInput';
 import Slider from '@/components/Slider';
 import Text from '@/components/Text';
+import { TextButton } from '@/components/Buttons';
 import TextInput from '@/components/TextInput';
+import { validateBirthYear, validateDisplayNameLength } from '../validators';
 
 import type { ApiMentor } from '@/features/MentorPage/mentorPageApi';
-import type { User } from '@/features/Authentication/authenticationApi';
 
 const PublicInfo = () => {
   const { t } = useTranslation('profile');
   const user = useAppSelector(selectUser);
   const mentor = useAppSelector(selectMentor);
-  const [updateUser] = useUpdateUserMutation();
-  const [updateMentor] = useUpdateMentorMutation();
-
-  const saveMentorData = useDebounce(
-    (mentor: ApiMentor) => updateMentor(mentor),
-    SAVE_DELAY_MS,
-  );
-
-  const saveUserAndMentorData = useDebounce((mentor: ApiMentor, user: User) => {
-    updateUser(user);
-    updateMentor(mentor);
-  }, SAVE_DELAY_MS);
+  const [updateMentor, { isLoading: isLoadingMentor }] =
+    useUpdateMentorMutation();
+  const [updateUser, { isLoading: isLoadingUser }] = useUpdateUserMutation();
 
   const [localData, setLocalData] = useState<ApiMentor>(mentor);
+  const [isDirty, setIsDirty] = useState(false);
   const [skillSearchValue, setSkillSearchValue] = useState('');
 
-  const handleDisplayNameChange = (display_name: string) => {
-    setLocalData({ ...localData, display_name });
-    saveUserAndMentorData(
-      { ...mentor, display_name },
-      { ...user, display_name },
-    );
-  };
-
   const updateMentorData = <K extends keyof ApiMentor>(
     key: K,
     value: ApiMentor[K],
   ) => {
     setLocalData({ ...localData, [key]: value });
-    saveMentorData({ ...mentor, [key]: value });
+    setIsDirty(true);
   };
 
+  const discardChanges = () => setLocalData(mentor);
+
+  const saveMentorData = () => {
+    // Only update user if display_name has changed
+    if (localData.display_name !== mentor.display_name) {
+      updateUser({ ...user, display_name: localData.display_name });
+    }
+    updateMentor(localData);
+  };
```

Comment:
Also we want to set isDirty to false, when we either discard changes or update the record

Reply:
(came to lurk here, wanted to see if this was maybe ready, because just heading for lunch)

Reply:
Yes, sorry. I had to go to the dentist so this was left unfinished. 😅

Reply:
Now fixed! 😃